modules/virt-accessing-vm-yaml-ssh.adoc

```diff
@@ -0,0 +1,168 @@
+        special: vm-ssh <2>
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: containerdisk
+          - disk:
+              bus: virtio
+            name: cloudinitdisk
+          interfaces:
+          - masquerade: {} <3>
+            name: testmasquerade <4>
+          rng: {}
+        machine:
+          type: ""
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+      - name: testmasquerade
+        pod: {}
+      volumes:
+      - name: containerdisk
+        containerDisk:
+          image: kubevirt/fedora-cloud-container-disk-demo
+      - name: cloudinitdisk
+        cloudInitNocloud:
+          userData: |
+            #!/bin/bash
+            echo "fedora" | passwd fedora --stdin
+...
+----
+<1> Name of the namespace created by the `oc create namespace` command.
+<2> Label used by the service to identify the virtual machine instances that are enabled for SSH traffic connections. The label can be any `key:value` pair that is added as a `label` to this YAML file and as a `selector` in the service YAML file.
+<3> The interface type is `masquerade`.
+<4> The name of this interface is `testmasquerade`.
+
+. Create the VM:
++
+[source,terminal,subs="+quotes"]
+----
+$ oc create -f __<path_for_the_VM_YAML_file>__
+----
++
+
+. Start the VM:
++
+[source,terminal]
+----
+$ virtctl start vm-ssh
+----
++
+
+. In the YAML file for the service, specify the service name, port number, and the target port.
+
++
+.Example output
+[source,yaml]
+----
+...
+apiVersion: v1
+kind: Service
+metadata:
+  name: svc-ssh <1>
+  namespace: ssh-ns <2>
+spec:
+  ports:
+  - targetPort: 22 <3>
+    protocol: TCP
+    port: 27017
+  selector:
+    special: vm-ssh <4>
+    type: NodePort
+...
+----
+<1> Name of the SSH service.
+<2> Name of the namespace created by the `oc create namespace` command.
+<3> The target port number for the SSH connection.
+<4> The selector name and value must match the label specified in the YAML file for the VM.
+
+. Create the service:
++
+[source,terminal,subs="+quotes"]
+----
+$ oc create -f __<path_for_the_service_YAML_file>__
+----
++
+. Verify that the VM is running:
++
+[source,terminal]
+----
+$ oc get vmi
+----
++
+.Example output
+[source,terminal]
+----
+NAME    AGE     PHASE       IP              NODENAME
+vm-ssh 6s       Running     10.244.196.152  node01
+----
+
+. Check the service to find out which port the service acquired:
++
+[source,terminal]
+----
+$ oc get svc
+----
++
+.Example output
+[source,terminal]
+----
+NAME            TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)           AGE
+svc-ssh     NodePort       10.106.236.208 <none>        27017:30093/TCP   22s
+----
++
+In this example, the service acquired the port number 30093.
+
+. Run the following command to obtain the IP address for the node:
++
+[source,terminal,subs"=+quotes"]
+----
+$ oc get node __<node_name>__ -o wide
+----
++
```

Comment:
Per your request in our session - this is the output of `oc get node <node name> -o wide`:
```
NAME     STATUS   ROLES    AGE     VERSION           INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                                                       KERNEL-VERSION                 CONTAINER-RUNTIME
node01   Ready    worker   6d22h   v1.20.0+5f82cdb   192.168.66.101   <none>        Red Hat Enterprise Linux CoreOS 48.83.202104040100-0 (Ootpa)   4.18.0-240.15.1.el8_3.x86_64   cri-o://1.21.0-53.rhaos4.8.gitec8716f.el8-dev
```
Please note that the IP address in this output is the one that is used in the `ssh` command.

Reply:
Added this output to the content. Please note that KERNEL-VERSION and CONTAINER-RUNTIME were excluded from the example output. Typically, we don't include those items in our example outputs.

Made a note of the IP address for the ssh command.